web/src/modules/topic/components/CriteriaTable/CriteriaTable.tsx

```diff
@@ -84,8 +89,10 @@ export const CriteriaTable = ({ problemNodeId }: Props) => {
   const criteria = nodeChildren.filter((node) => node.type === "criterion");
   const solutions = nodeChildren.filter((node) => node.type === "solution");
 
-  const rowData = buildRows(criteria, solutions, criterionSolutionEdges);
-  const columns = buildColumns(solutions);
+  const rowData = isTableTransposed
+    ? buildRows(solutions, criteria, criterionSolutionEdges)
```

Comment:
I think this can be slightly cleaner/clearer by doing
```
const rowNodes = useSolutionsForColumns ? criteria : solutions;
const columnNodes = useSolutionsForColumns ? solutions : criteria;
```
then `buildRows` and `buildColumns` can be invoked unconditionally with these variables